pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/nullvalue/RealtimeNullValueVectorReaderWriter.java

```diff
@@ -39,4 +40,9 @@ public void setNull(int docId) {
   public boolean isNull(int docId) {
     return _nullBitmap.contains(docId);
   }
+
+  @Override
+  public ImmutableRoaringBitmap getNullBitmap() {
```

Comment:
@Jackie-Jiang why do we need that? there is only one thread updating the bitmap and multiple threads reading it. cloning entire bitmap for every access is going to be expensive. when did we do this change?

Reply:
@kishoreg MutableRoaringBitmap itself is not thread-safe. You cannot read the bitmap while updating it. One possible solution is using ReadWriteLock for this.

Reply:
Actually ReadWriteLock might not work in this case as there are too many reading methods

Reply:
I didn't know about this. we should measure the performance impact on the ingestion rate (due to synchronization) and query throughput (bcos to clone). Readwrite lock will not help since its getting updated frequently.

 The synchronization is not really needed since only one thread(consumer) is updating it. clone is important.

Reply:
We need synchronization because clone and update cannot happen at the same time either

Reply:
Missed that. let's create another issue to measure and optimize this.